ai_agents/devable_research_agent/scripts/run_pipeline.py

```
@@ -0,0 +1,137 @@
+    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    line = f"[{timestamp}] {message}"
+    with LOG_FILE.open("a", encoding="utf-8") as handle:
+        handle.write(line + "\n")
+    print(line)
+
+
+def _run_step(name: str, cmd: list) -> bool:
+    _log(f"START {name}")
+    _log(f"CMD {' '.join(cmd)}")
+    try:
+        result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)